consumer/consumermock/mock_consumers.go

```diff
@@ -0,0 +1,197 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumermock
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/atomic"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
+)
+
+var _ consumer.TraceConsumer = (*Trace)(nil)
+var _ consumer.TraceConsumerOld = (*Trace)(nil)
+
+type Trace struct {
```

Comment:
This is very similar with https://github.com/open-telemetry/opentelemetry-collector/blob/master/exporter/exportertest/sink_exporter.go. We should try to avoid having duplicate code like this.

Reply:
Yeah the goal is to get rid of it.